packages/core/src/components/fields/ObjectField.js

```diff
@@ -83,7 +83,7 @@ class ObjectField extends Component {
 
   onPropertyChange = (name, addedByAdditionalProperties = false) => {
     return (value, errorSchema) => {
-      if (!value && addedByAdditionalProperties) {
+      if (typeof value === "undefined" && addedByAdditionalProperties) {
```

Comment:
This change seems unrelated to the oneOf change -- could you add another test to cover it? (I believe this fixes an issue in which you can't type in "0" in a oneOf textbox, I think)